libs/core/version/src/version.cpp

```diff
@@ -131,6 +131,16 @@ namespace hpx {
     }
 #endif
 
+#if (defined(HPX_HAVE_NETWORKING) && defined(HPX_HAVE_PARCELPORT_GASNET)) ||   \
+    defined(HPX_HAVE_MODULE_GASNET_BASE)
+    std::string gasnet_version()
+    {
+        std::ostringstream strm;
+        strm << "GASNET";
```

Comment:
@ct-clmsn is there a way to be more specific here?

Reply:
You probably want these defines provided by the GASNet headers:

- `GASNET_RELEASE_VERSION_MAJOR`
- `GASNET_RELEASE_VERSION_MINOR` 
- `GASNET_RELEASE_VERSION_PATCH` 

Or possibly just `GASNET_CONFIG_STRING` if you want a human-readable string containing the version and lots of other useful configuration details

Reply:
@hkaiser sure thing; in `../include/gasnet.h` there are major an minor versions defined.
```
/* public spec version numbers */
#define GASNET_SPEC_VERSION_MAJOR GASNETI_SPEC_VERSION_MAJOR
#define GASNET_SPEC_VERSION_MINOR GASNETI_SPEC_VERSION_MINOR

/*  legacy name for major spec version number */
#define GASNET_VERSION GASNET_SPEC_VERSION_MAJOR
```

Reply:
@ct-clmsn `GASNET_SPEC_VERSION_{MAJOR,MINOR}` are probably **NOT** what you want. Those are the spec versions for the (now frozen) deprecated GASNet-1 API, which has not changed since 2006

The symbols I gave in my comment are better choices.

Reply:
@bonachea thanks for the clarification!

Reply:
@hkaiser `GASNET_CONFIG_STRING` gets into build options: version of gasnet, thread model, cache line byte count, debug build status, etc - looks like a reasonable choice. (@bonachea thank you for the suggestion)

Reply:
FWIW, this is printed by `--hpx:version` and/or `--hpx:info`.

Reply:
@hkaiser version numbers would be sufficient?

Reply:
Version numbers should be fine, perhaps we could also add the conduit used.

Reply:
I'll work on this and @bonachea's corrections.